Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -0,0 +1,282 @@
+                return CreatedAtAction("AddParticipantToTeam", participantTeam);
+            }
+            else
+                return UnprocessableEntity($"Participant {participant.Username} already is a part of a team in this activity");
+        }
+
+        /// <summary>
+        /// Updates Participant role in a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="participant">Default Role Value value 3 (Member)</param>
+        /// <returns></returns>
+        [HttpPatch]
+        [Route("{teamID}/participants")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.RECIM_TEAM)]
+        public async Task<ActionResult<ParticipantTeamViewModel>> UpdateParticipantTeam(int teamID, ParticipantTeamUploadViewModel participant)
+        {
+            
+                participant.RoleTypeID = participant.RoleTypeID ?? 3;
+                var participantTeam = await _teamService.AddParticipantToTeamAsync(teamID, participant);
+                return CreatedAtAction("AddParticipantToTeam", participantTeam);
+            
+        }
+
+        /// <summary>
+        /// Removes specified user from a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="username"></param>
+        /// <returns></returns>
+        [HttpDelete]
+        [Route("{teamID}/participants")]
+        public async Task<ActionResult> DeleteTeamParticipant(int teamID, string username)
+        {
+            var user_name = AuthUtils.GetUsername(User);
+            var participantTeam = _teamService.GetParticipantTeam(teamID, username);
+            if (participantTeam is null)
+                return NotFound("The user is not part of the team.");
+            if (user_name != participantTeam.ParticipantUsername)
+                return Forbid($"You are not permitted to reject invitations for another participant.");
+
+            await _teamService.DeleteParticipantTeamAsync(teamID, username);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Update a team info
+        /// </summary>
+        /// <param name="team"></param>
+        /// <param name="teamID"></param>
+        /// <returns></returns>
+        [HttpPatch]
+        [Route("{teamID}")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.RECIM_TEAM)]
+        public async Task<ActionResult<TeamViewModel>> UpdateTeamInfo(int teamID, TeamPatchViewModel team)
+        {
+            if (team.Name is not null)
+            {
+                var activityID = _teamService.GetTeamByID(teamID).Activity.ID;
+                if (_teamService.HasTeamNameTaken(activityID, team.Name))
+                    return UnprocessableEntity($"Team name {team.Name} has already been taken by another team in this activity");
+
+            }
+
+            var updatedTeam = await _teamService.UpdateTeamAsync(teamID, team);
```

Comment:
I suppose this is more of a "I'm not sure how to properly specify an error". When attempting to return something specific like an UnprocessableEntity, c# automatically suggests that I pass the exception as a parameter

Reply:
I'm not fully sure how to return a specific error message like an UnprocessableEntity. However, my new implementation simply throws and catches an exception. Is there a better way of handling this?

Reply:
Throwing and then handling exceptions is the standard error control flow in C#.

The only "better" way of handling it (which is still arguable), is to add a new exception type to our `CustomExceptions.cs` file, which transforms exceptions into HTTP Responses with a Status Code corresponding to the specific exception thrown. This way, you can throw the exception where the error is detected and, if no one else catches, the API will automatically transform it into the appropriate HTTP Response rather than a generic `500 Internal Server Error`.

Reply:
I like this, I'll add UnprocessibleEntity to our custom exceptions if that is okay.